src/core/tsi/gts_transport_security.c

```diff
@@ -0,0 +1,35 @@
+/*
+ *
+ * Copyright 2017 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "src/core/tsi/gts_transport_security.h"
+#include <grpc/support/alloc.h>
+
+static gts_shared_resource* gts_resource = NULL;
```

Comment:
The pointer will be set during the time gts tsi handshaker is created (in a lazy manner), and I think the main reason is to save resources i.e., if there is no handshake performed, we will not create any shared resources including tsi thread, cq and so on.  How do you think?